<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindowClass</class>
 <widget class="QMainWindow" name="MainWindowClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>600</width>
    <height>529</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Approximator</string>
  </property>
  <property name="windowIcon">
   <iconset resource="aproxim.qrc">
    <normaloff>:/new/prefix1/images/sigma.ico</normaloff>:/new/prefix1/images/sigma.ico</iconset>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout_2">
    <item row="0" column="0" colspan="2">
     <widget class="QTabWidget" name="tabWidget">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="tab">
       <attribute name="title">
        <string>Parameters</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout">
        <item row="0" column="0" colspan="2">
         <widget class="QLineEdit" name="value_x">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>X value</string>
          </property>
         </widget>
        </item>
        <item row="0" column="2">
         <widget class="QLineEdit" name="value_y">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>Y value</string>
          </property>
         </widget>
        </item>
        <item row="0" column="3" colspan="2">
         <widget class="QSpinBox" name="selected_precision">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimum">
           <number>1</number>
          </property>
          <property name="value">
           <number>4</number>
          </property>
         </widget>
        </item>
        <item row="0" column="5">
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>266</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="textLabel1">
          <property name="text">
           <string>Select function</string>
          </property>
          <property name="wordWrap">
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item row="1" column="1" colspan="3">
         <widget class="QComboBox" name="selected_function">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <item>
           <property name="text">
            <string>Y=A+B*X</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Y=A+B/X</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Y=A*X^B</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Y=10^(A+B*X)</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Y=EXP(A+B*X)</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Y=A*EXP(B*X)</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Y=A*B^X</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Y=A+B*LOG10(X)</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Y=A+B*LOG(X)</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Y=X/(A+B*X)</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Y=A*X/(B+X)</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Y=1/(A+B*X)</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Y=A*10^(B*X)</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Y=1/(A+B*EXP(-X))</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Y=A/(B+X)</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Y=A*EXP(B/X)</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Y=A*10^(B/X)</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Y=A+B*X^n</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Y=Pn(X)</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Y=A*X^B*EXP(B2*X)</string>
           </property>
          </item>
         </widget>
        </item>
        <item row="1" column="4" colspan="2">
         <spacer name="horizontalSpacer_2">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>297</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="2" column="0" colspan="3">
         <widget class="QLabel" name="value_a">
          <property name="text">
           <string>A = </string>
          </property>
          <property name="wordWrap">
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item row="3" column="0" colspan="6">
         <widget class="QLabel" name="value_b">
          <property name="text">
           <string>B = </string>
          </property>
          <property name="wordWrap">
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item row="4" column="0" colspan="6">
         <widget class="QLabel" name="value_b2">
          <property name="text">
           <string>B2 = </string>
          </property>
          <property name="wordWrap">
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item row="5" column="0" colspan="6">
         <widget class="QLabel" name="value_ad">
          <property name="text">
           <string>Average deviation: </string>
          </property>
          <property name="wordWrap">
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item row="6" column="0" colspan="6">
         <widget class="QLabel" name="value_aqd">
          <property name="text">
           <string>Mean-square deviation: </string>
          </property>
          <property name="wordWrap">
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item row="7" column="0" colspan="6">
         <widget class="QLabel" name="value_cc">
          <property name="text">
           <string>Correlation coefficient: </string>
          </property>
          <property name="wordWrap">
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item row="8" column="0">
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>141</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_2">
       <attribute name="title">
        <string>Data</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_3">
        <item row="0" column="0">
         <widget class="QTableWidget" name="data_table">
          <property name="sortingEnabled">
           <bool>true</bool>
          </property>
          <property name="rowCount">
           <number>10</number>
          </property>
          <property name="columnCount">
           <number>4</number>
          </property>
          <attribute name="horizontalHeaderShowSortIndicator" stdset="0">
           <bool>true</bool>
          </attribute>
          <attribute name="horizontalHeaderStretchLastSection">
           <bool>true</bool>
          </attribute>
          <attribute name="verticalHeaderShowSortIndicator" stdset="0">
           <bool>false</bool>
          </attribute>
          <attribute name="verticalHeaderStretchLastSection">
           <bool>false</bool>
          </attribute>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <column>
           <property name="text">
            <string>X</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Y0</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Y</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>%</string>
           </property>
          </column>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_3">
       <attribute name="title">
        <string>Graph</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_4">
        <item row="0" column="0">
         <widget class="QtCharts::QChartView" name="data_plot"/>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionNew"/>
   <addaction name="actionOpen"/>
   <addaction name="actionSave"/>
   <addaction name="actionApproximation"/>
   <addaction name="actionCalculatePoint"/>
   <addaction name="actionDifferential"/>
   <addaction name="actionIntegral"/>
   <addaction name="actionFindModel"/>
   <addaction name="actionOptions"/>
   <addaction name="actionAbout"/>
   <addaction name="actionExit"/>
  </widget>
  <action name="actionNew">
   <property name="icon">
    <iconset resource="aproxim.qrc">
     <normaloff>:/new/prefix1/images/new.png</normaloff>:/new/prefix1/images/new.png</iconset>
   </property>
   <property name="text">
    <string>New</string>
   </property>
   <property name="toolTip">
    <string>New</string>
   </property>
  </action>
  <action name="actionOpen">
   <property name="icon">
    <iconset resource="aproxim.qrc">
     <normaloff>:/new/prefix1/images/open.png</normaloff>:/new/prefix1/images/open.png</iconset>
   </property>
   <property name="text">
    <string>Open</string>
   </property>
   <property name="toolTip">
    <string>Open</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="icon">
    <iconset resource="aproxim.qrc">
     <normaloff>:/new/prefix1/images/save.png</normaloff>:/new/prefix1/images/save.png</iconset>
   </property>
   <property name="text">
    <string>Save</string>
   </property>
   <property name="toolTip">
    <string>Save</string>
   </property>
  </action>
  <action name="actionApproximation">
   <property name="icon">
    <iconset resource="aproxim.qrc">
     <normaloff>:/new/prefix1/images/run.png</normaloff>:/new/prefix1/images/run.png</iconset>
   </property>
   <property name="text">
    <string>Approximation</string>
   </property>
   <property name="toolTip">
    <string>Approximation</string>
   </property>
  </action>
  <action name="actionCalculatePoint">
   <property name="icon">
    <iconset resource="aproxim.qrc">
     <normaloff>:/new/prefix1/images/point.png</normaloff>:/new/prefix1/images/point.png</iconset>
   </property>
   <property name="text">
    <string>Calculate point</string>
   </property>
   <property name="toolTip">
    <string>Calculate point</string>
   </property>
  </action>
  <action name="actionDifferential">
   <property name="icon">
    <iconset resource="aproxim.qrc">
     <normaloff>:/new/prefix1/images/differential.png</normaloff>:/new/prefix1/images/differential.png</iconset>
   </property>
   <property name="text">
    <string>Differential</string>
   </property>
   <property name="toolTip">
    <string>Differential</string>
   </property>
  </action>
  <action name="actionIntegral">
   <property name="icon">
    <iconset resource="aproxim.qrc">
     <normaloff>:/new/prefix1/images/integral.png</normaloff>:/new/prefix1/images/integral.png</iconset>
   </property>
   <property name="text">
    <string>Integral</string>
   </property>
   <property name="toolTip">
    <string>Integral</string>
   </property>
  </action>
  <action name="actionFindModel">
   <property name="icon">
    <iconset resource="aproxim.qrc">
     <normaloff>:/new/prefix1/images/model.png</normaloff>:/new/prefix1/images/model.png</iconset>
   </property>
   <property name="text">
    <string>Try to find model</string>
   </property>
   <property name="toolTip">
    <string>Try to find model</string>
   </property>
  </action>
  <action name="actionOptions">
   <property name="icon">
    <iconset resource="aproxim.qrc">
     <normaloff>:/new/prefix1/images/settings.png</normaloff>:/new/prefix1/images/settings.png</iconset>
   </property>
   <property name="text">
    <string>Options</string>
   </property>
   <property name="toolTip">
    <string>Options</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="icon">
    <iconset resource="aproxim.qrc">
     <normaloff>:/new/prefix1/images/about.png</normaloff>:/new/prefix1/images/about.png</iconset>
   </property>
   <property name="text">
    <string>About</string>
   </property>
   <property name="toolTip">
    <string>About</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="icon">
    <iconset resource="aproxim.qrc">
     <normaloff>:/new/prefix1/images/quit.png</normaloff>:/new/prefix1/images/quit.png</iconset>
   </property>
   <property name="text">
    <string>Exit</string>
   </property>
   <property name="toolTip">
    <string>Exit</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>QtCharts::QChartView</class>
   <extends>QGraphicsView</extends>
   <header location="global">QtCharts/QChartView&gt;
#include &lt;QtCharts/chartsnamespace.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="aproxim.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>actionAbout</sender>
   <signal>triggered()</signal>
   <receiver>MainWindowClass</receiver>
   <slot>aboutClick()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>307</x>
     <y>49</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionExit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindowClass</receiver>
   <slot>quitClick()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>379</x>
     <y>42</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionOptions</sender>
   <signal>triggered()</signal>
   <receiver>MainWindowClass</receiver>
   <slot>optionsClick()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>445</x>
     <y>41</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionFindModel</sender>
   <signal>triggered()</signal>
   <receiver>MainWindowClass</receiver>
   <slot>findModelClick()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>528</x>
     <y>43</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionIntegral</sender>
   <signal>triggered()</signal>
   <receiver>MainWindowClass</receiver>
   <slot>integralClick()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>209</x>
     <y>39</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionDifferential</sender>
   <signal>triggered()</signal>
   <receiver>MainWindowClass</receiver>
   <slot>differentialClick()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>180</x>
     <y>43</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionCalculatePoint</sender>
   <signal>triggered()</signal>
   <receiver>MainWindowClass</receiver>
   <slot>calculatePointClick()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>151</x>
     <y>38</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionApproximation</sender>
   <signal>triggered()</signal>
   <receiver>MainWindowClass</receiver>
   <slot>approximateClick()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>116</x>
     <y>39</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSave</sender>
   <signal>triggered()</signal>
   <receiver>MainWindowClass</receiver>
   <slot>saveClick()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>89</x>
     <y>38</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionOpen</sender>
   <signal>triggered()</signal>
   <receiver>MainWindowClass</receiver>
   <slot>openClick()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>58</x>
     <y>40</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionNew</sender>
   <signal>triggered()</signal>
   <receiver>MainWindowClass</receiver>
   <slot>newClick()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>28</x>
     <y>39</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>aboutClick()</slot>
  <slot>quitClick()</slot>
  <slot>newClick()</slot>
  <slot>openClick()</slot>
  <slot>saveClick()</slot>
  <slot>approximateClick()</slot>
  <slot>calculatePointClick()</slot>
  <slot>findModelClick()</slot>
  <slot>optionsClick()</slot>
  <slot>integralClick()</slot>
  <slot>differentialClick()</slot>
 </slots>
</ui>
